app/main/views/users.py

```diff
@@ -271,24 +282,44 @@ def export_users_for_framework(framework_slug):
 
     suppliers_with_a_complete_service = frozenset(framework.get_supplier_ids_for_completed_service())
 
-    supplier_id_published_service_count = dict(db.session.query(
-        Service.supplier_id,
-        func.count(Service.id)
-    ).filter(
-        Service.status == 'published',
-        Service.framework_id == framework.id
-    ).group_by(
-        Service.supplier_id
-    ).all())
+    lots_ids = db.session.query(FrameworkLot.lot_id).filter(FrameworkLot.framework_id == framework.id).all()
+    lots = db.session.query(Lot).filter(Lot.id.in_(lots_ids)).all()
+    published_service_count_by_supplier_and_lot = {}
+
+    for lot in lots:
+        published_service_count_by_supplier_and_lot[
+            "published_services_count_on_{}_lot".format(lot.slug)
+        ] = dict(db.session.query(
+            Service.supplier_id,
+            func.count(Service.id)
+        ).filter(
+            Service.status == 'published',
+            Service.lot_id == lot.id
+        ).group_by(
+            Service.supplier_id
+        ).group_by(
+            Service.lot_id
+        ).all())
+
+    supplier_id_published_service_count = {}
+    for k, v in published_service_count_by_supplier_and_lot.items():
+        for supplier_id, service_count in v.items():
+            if supplier_id in supplier_id_published_service_count:
+                supplier_id_published_service_count[supplier_id] += service_count
+            else:
+                supplier_id_published_service_count[supplier_id] = service_count
+
 
     supplier_frameworks_and_users = db.session.query(
-        SupplierFramework, User
+        SupplierFramework, User, ContactInformation
```

Comment:
It might be better to explicitly add `Supplier` here too. Accessing it though the attribute on user _could_ have some undesirable effects such as calling the DB once per supplier framework to find the corresponding supplier object.

This depends if the Supplier object is lazy loaded with the user. It might be fine.
Is there a specific reason you added `ContactInformation` and not `Supplier`? You should be able to do `u.supplier.contact_information.phone_number` for example (not that I'd recommend that, just curious)

Reply:
As per our chat a few mins ago, supplier is lazy loaded on the user indeed. And just so we remember I am adding here that we can check if contact information is lazy loaded together with the supplier :).